### PR TITLE
add clang-format

### PR DESCRIPTION
**Pull request by nomaddo**
- check sources are formatted in CI
- add clang-format command to cmake

Discussed in #57.
If new commits are not formatted by clang-format, `check-code-style` failed in CI.
The format style is now created just by `clang-format -dump-config -style=LLVM > .clang-format`.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -75,6 +75,24 @@ set_target_properties(
 	SOVERSION "1.2"
 )
 
+find_program(
+  CLANG_PATH
```
Comment (doe300) [inline src/CMakeLists.txt]: You are overwriting `CLANG_PATH` here. I am not sure, if CMake distinguishes between variables set in different directories.
To be on the safe side, please rename to e.g. `CLANG_FORMAT_PATH`.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -75,6 +75,24 @@ set_target_properties(
...
+  clang-format
+  )
+
+if(CLANG_PATH)
+  message (STATUS "found clang-format: ${CLANG_PATH}")
+  add_custom_target(
+    clangformat
+    COMMAND ${CLANG_PATH}
+    -style=LLVM
```
Comment (doe300) [inline src/CMakeLists.txt]: Doesn't this need to be `-style=file` to use the style from the `.clang-format` file?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -75,6 +75,24 @@ set_target_properties(
...
+    -i
+    ${SRCS}
+    )
+else()
+  message (WARNING "clang-format not found: strongly recoomend to use it before commit!")
```
Comment (doe300) [inline src/CMakeLists.txt]: typo: recoomend -> recommend

File: src/CMakeLists.txt (hunk 2 above)
Comment (nomaddo) [reply src/CMakeLists.txt]: If the parent directories has `.clang-format`, we don't need to speciy it.

**Comment by nomaddo** [comment]
Fixed. If your like the coding rules, please commit about formatting.

**Comment by doe300** [comment]
I created 2 gists with the default LLVM style [here](https://gist.github.com/doe300/892e118f6c9527cfc886651650bf12c7) and with my own personal configuration [here](https://gist.github.com/doe300/6517268d4a6d8f02e5edfc85b08d1de9).
Any comments are appreciated.

if you want to play with the settings a bit, I found [this here](https://zed0.co.uk/clang-format-configurator/).

**Comment by nomaddo** [comment]
Your config seems good for me.
I like spaces instead of tabs, but it's not a problem.

**Pull request merged**